test/eigsolve.jl

```diff
@@ -342,20 +342,290 @@ end
+        alg2 = BlockLanczos(; krylovdim=2n, maxiter=10, tol=tolerance(T),
+                            verbosity=WARN_LEVEL)
+        evals1, _, info1 = eigsolve(wrapop(A, Val(mode)), x₀_lanczos, n, :SR, alg1)
+        evals2, _, info2 = eigsolve(wrapop(A, Val(mode)), x₀_block, n, :SR, alg2)
+        @test info1.converged == info2.converged
+    end
+    @testset for T in scalartypes
+        A = rand(T, (2N, 2N)) .- one(T) / 2
+        A = (A + A') / 2
+        block_size = 4
+        x₀_block = Block{T}([wrapvec(rand(T, 2N), Val(mode)) for _ in 1:block_size])
+        x₀_lanczos = x₀_block[1]
+        alg1 = Lanczos(; krylovdim=2n, maxiter=10, tol=tolerance(T), verbosity=WARN_LEVEL)
+        alg2 = BlockLanczos(; krylovdim=2n, maxiter=10, tol=tolerance(T),
+                            verbosity=WARN_LEVEL)
+        evals1, _, info1 = eigsolve(wrapop(A, Val(mode)), x₀_lanczos, n, :SR, alg1)
+        evals2, _, info2 = eigsolve(wrapop(A, Val(mode)), x₀_block, n, :SR, alg2)
+        @test info1.converged >= info2.converged + 1
+    end
+end
+
+# Test effectiveness of shrink!() in BlockLanczos
+@testset "Test effectiveness of shrink!() in BlockLanczos $mode" for mode in
+                                                                     (:vector, :inplace,
+                                                                      :outplace)
+    scalartypes = mode === :vector ? (Float32, Float64, ComplexF32, ComplexF64) :
+                  (ComplexF64,)
+    @testset for T in scalartypes
+        block_size = 5
+        A = mat_with_eigrepition(T, N, block_size)
+        x₀ = Block{T}([wrapvec(rand(T, N), Val(mode)) for _ in 1:block_size])
+        values0 = eigvals(A)[1:n]
+        n1 = n ÷ 2
+        alg = BlockLanczos(; krylovdim=3 * n ÷ 2, maxiter=1, tol=1e-12)
+        values, _, _ = eigsolve(wrapop(A, Val(mode)), x₀, n, :SR, alg)
+        error1 = norm(values[1:n1] - values0[1:n1])
+        alg_shrink = BlockLanczos(; krylovdim=3 * n ÷ 2, maxiter=2, tol=1e-12)
+        values_shrink, _, _ = eigsolve(wrapop(A, Val(mode)), x₀, n, :SR, alg_shrink)
+        error2 = norm(values_shrink[1:n1] - values0[1:n1])
+        @test error2 < error1
+    end
+end
+
+@testset "BlockLanczos - eigsolve without alg $mode" for mode in
+                                                         (:vector, :inplace, :outplace)
+    scalartypes = mode === :vector ? (Float32, Float64, ComplexF32, ComplexF64) :
+                  (ComplexF64,)
+    @testset for T in scalartypes
+        block_size = 2
+        A = mat_with_eigrepition(T, n, block_size)
+        x₀ = Block{T}([wrapvec(rand(T, n), Val(mode)) for _ in 1:block_size])
+        if mode === :vector
+            D1, V1, info1 = eigsolve(wrapop(A, Val(mode)), x₀, 1, :SR)
+            eigA = eigvals(A)
+            @test D1[1] ≈ eigA[1]
+            D2, V2, info2 = eigsolve(wrapop(A, Val(mode)), x₀)
+            D2[1] = max(abs(eigA[1]), abs(eigA[end]))
```

Comment:
Was this supposed to be a test?
```suggestion
            @test D2[1] ≈ max(abs(eigA[1]), abs(eigA[end]))
```

Reply:
Yes, this was intended to be a test. I've now revised it accordingly to
```julia
@test D2[1] ≈ (abs(eigA[1]) > abs(eigA[end]) ? eigA[1] : eigA[end])
```